### PR TITLE
Split guests

### PR DESCRIPTION
**Pull request by aanisov**
Make configurable set of guests to be built and installed for this product.

### CODE REVIEW

File: recipes-dom0/dom0-image-thin-initramfs/dom0-image-thin-initramfs.bbappend [hunk 1]
```diff
@@ -28,6 +37,7 @@ XT_BB_LOCAL_CONF_FILE = "meta-xt-prod-extra/doc/local.conf.dom0-image-minimal-in
 
 XT_BB_IMAGE_TARGET = "core-image-thin-initramfs"
 
+
```
Comment (andr2000) [inline recipes-dom0/dom0-image-thin-initramfs/dom0-image-thin-initramfs.bbappend]: not needed

File: recipes-dom0/dom0-image-thin-initramfs/dom0-image-thin-initramfs.bbappend [hunk 2]
```diff
@@ -2,8 +2,17 @@ FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
...
 do_fetch[depends] += "domd-agl-demo-platform:do_${BB_DEFAULT_TASK}"
-do_fetch[depends] += "domu-image-android:do_${BB_DEFAULT_TASK}"
-do_fetch[depends] += "domu-image-fusion:do_${BB_DEFAULT_TASK}"
+
+XT_GUESTS_BUILD ?= "doma domf"
+XT_GUESTS_INSTALL ?= "doma domf"
+
+python __anonymous () {
+    guests = d.getVar('XT_GUESTS_BUILD', True).split()
+    if "doma" in guests :
```
Comment (andr2000) [inline recipes-dom0/dom0-image-thin-initramfs/dom0-image-thin-initramfs.bbappend]: I would better have not an "add domain option", but rather "remove domain from the build", e.g. you have a default set of domains to be built and by remove option you can remove those you don't want.
So, the default product builds as usually

Comment (aanisov) [reply recipes-dom0/dom0-image-thin-initramfs/dom0-image-thin-initramfs.bbappend]: A line above this function is defined a default set of domains. In my case I want to set `XT_GUESTS_BUILD = "doma domr"`, in such a case 'removing from default' semantic would look weird.

Comment (andr2000) [reply recipes-dom0/dom0-image-thin-initramfs/dom0-image-thin-initramfs.bbappend]: ok, my bad

File: recipes-dom0/dom0-image-thin-initramfs/dom0-image-thin-initramfs.bbappend (hunk 1 above)
Comment (aanisov) [reply recipes-dom0/dom0-image-thin-initramfs/dom0-image-thin-initramfs.bbappend]: yep

**Comment by aanisov** [comment]
Wiped out empty line change, added a bit more details to commit message, added RB from @andr2000 .
I guess this is ready to be merged.

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/xt_set_root_dev_cfg.sh [hunk 3]
```diff
@@ -27,4 +27,7 @@ STORAGE_PART="${BOOT_STORAGE}${PART_PREFIX}"
 # now make up the configuration
 echo "Mangling domain configuration: setting storage to ${BOOT_STORAGE}"
 sed -i "s/STORAGE_PART/${STORAGE_PART}/g" ${DOM_CFG_DIR}/domd.cfg
-sed -i "s/STORAGE_PART/${STORAGE_PART}/g" ${DOM_CFG_DIR}/domf.cfg
+
+if [ -f /xt/dom.cfg/domf.cfg ] ; then
+    sed -i "s/STORAGE_PART/${STORAGE_PART}/g" ${DOM_CFG_DIR}/domf.cfg
+fi
```
Comment (aanisov) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/xt_set_root_dev_cfg.sh]: @andr2000 Please take a look at this additional change in this PR.

Comment (aanisov) [reply recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/xt_set_root_dev_cfg.sh]: sorry, I'll fix this now

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/xt_set_root_dev_cfg.sh [hunk 4]
```diff
@@ -27,4 +27,7 @@ STORAGE_PART="${BOOT_STORAGE}${PART_PREFIX}"
 # now make up the configuration
 echo "Mangling domain configuration: setting storage to ${BOOT_STORAGE}"
 sed -i "s/STORAGE_PART/${STORAGE_PART}/g" ${DOM_CFG_DIR}/domd.cfg
-sed -i "s/STORAGE_PART/${STORAGE_PART}/g" ${DOM_CFG_DIR}/domf.cfg
+
+if [ -f ${DOM_CFG_DIR}/domd.cfg ] ; then
+    sed -i "s/STORAGE_PART/${STORAGE_PART}/g" ${DOM_CFG_DIR}/domf.cfg
+fi
```
Comment (aanisov) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/xt_set_root_dev_cfg.sh]: @andr2000  please take a look

**Pull request merged**